### ISSUE TITLE
stringify() produces invalid result when object contains "length" property

### ISSUE THREAD

**Issue by tony-vu** [mongodb-js/extended-json#20]
When calling EJSON.stringify() on an object that contains a "length" property, an invalid result is returned.
Yes, I know it's bad practice to have a property named "length", and worst yet if the value is a number!! 
Unfortunately it's not within my control, because the object is coming back from mongodb's [GridFS files collection](https://docs.mongodb.org/manual/reference/gridfs/#gridfs-files-collection).

Here's a example:
```javascript
var ObjectId = require('mongodb').ObjectId,
    EJSON = require('mongodb-extended-json');

var doc = [{
    "_id": new ObjectId("562555959995d0e81a76ad08"),
    "filename": "sample-0.1.0.tgz",
    "contentType": "application/x-compressed",
    "length": 765,
    "chunkSize": 261120,
    "uploadDate": new Date("2015-10-19T20:41:57.273Z"),
    "aliases": null,
    "metadata": {},
    "md5": "f70be8b905480bc0f25214d05a8e3a24"
}];
console.log(EJSON.stringify(doc));

// Output: 
// [{"0":{"$undefined":true},"1":{"$undefined":true},"2":{"$undefined":true},"3":{"
// $undefined":true},"4":{"$undefined":true},"5":{"$undefined":true},"6":{"$undefin
// ed":true},"7":{"$undefined":true},"8":{"$undefined":true},"9":{"$undefined":true
// ....
```

The source of the problem is that the `inflate()` method eventually calls `lodash.reduce` which interprets the input as an "array-like" object and iterates over it using the length value.

I'm working-around this issue by replacing `length` with an alternate name and then restoring it later.

**Comment by Machyne** [comment]
It seems that this is a lodash issue. :)
I reported it here: https://github.com/lodash/lodash/issues/1705

**Comment by Machyne** [comment]
Never mind! Looks like this package just needed to use a different lodash method.

**Comment by tony-vu** [comment]
@Machyne thanks for the fix. I tried using `_.transform` and it does indeed work. I didn't know about that subtlety of `reduce` vs `transform`. Thanks @jdalton for that insight.

**Comment by jdalton** [comment]
Np ;D

**Comment by imlucas** [comment]
published in mongodb-extended-json@v1.5.6
